# Instructions
Below is a list of elements where each element is a different data type. Your task is to perform some basic analysis on the list of elements to come up with the following report.

1. What is the data type of the element in position 3?
2. are the data types in position 0 and position 2 the same?
3. index the list so that you return the tuple. Assign the tuple to a new variable called `extracted`. What is the data type of the element in the 2nd position of this tuple?
4. Index the list to return the string `return me`, what are the characters at position 0, 3, and 5
5. Index the list to return the string `return me`, concatenate the characters at positions 0, 3, and 5 to create a new string. What is the new string?
6. return the element at position 9 and store it in a value called `squares`. What is the data type and what are it's values.
7. What is the result of the element at position 1 subtracted from the element at position 10
8. what is the memory address of the list `data`
9. are the memory addresses of the element in position 2 and position 8 the same? ***Bonus: If they are the same, explain why***

In [4]:
data = [
    True,
    600.548,
    False,
    45j,
    [1,2,3,4,5],
    False,
    (True, "Minecraft is the best", 10.098, 'one billion', {1,2,3,4}),
    "return me",
    5 == 5,
    [x**2 for x in [1,2,3,4,5]],
    654.548    
]

In [28]:
#1
print("1", type(data[3]))
#2
print("2",bool(type(data[0] == data[2])))
#3
extracted = data[6]
print("3 extracted value:", extracted, "Data type of 2nd element:", type(extracted[1]))
#4
thing = data[7]
print("4 ", thing[0], thing[3], thing[5])
#5
thing = data[7]
new = thing[0]+thing[3]+thing[5]
print("5", new)
#6
squares = data[9]
print("6 ", type(squares), squares)
#7
result = data[1] - data[10]
print("7", result)
#8
print("8 ", id(data))
#9
same = id(data[2]) == id(data[8])
print("9", same)

1 <class 'complex'>
2 True
3 extracted value: (True, 'Minecraft is the best', 10.098, 'one billion', {1, 2, 3, 4}) Data type of 2nd element: <class 'str'>
4  r u n
5 run
6  <class 'list'> [1, 4, 9, 16, 25]
7 -54.0
8  140635917661440
9 False


## YOUR ANSWERS GO HERE
1. Complex
2. Yes they are both boolean
3. The second element in extracted is string
4. The characters are r, u, n.
5. new string is 'run'
6. It is a list. Its values are 1,4,9,16,25
7. result is -54.0
8. memory address of list data is 140635917661440
9. They do not have the same memory address



